services/pull/check.go

```diff
@@ -176,40 +178,45 @@ func TestPullRequests() {
 		}
 
 		checkAndUpdateStatus(pr)
+		select {
+		case <-ctx.Done():
+			log.Info("PID: %d Pull Request testing shutdown", os.Getpid())
+			return
+		default:
+		}
 	}
 
 	// Start listening on new test requests.
-	for prID := range pullRequestQueue.Queue() {
-		log.Trace("TestPullRequests[%v]: processing test task", prID)
-		pullRequestQueue.Remove(prID)
-
-		id := com.StrTo(prID).MustInt64()
-		if _, ok := checkedPRs[id]; ok {
-			continue
-		}
+	for {
+		select {
+		case prID := <-pullRequestQueue.Queue():
+			log.Trace("TestPullRequests[%v]: processing test task", prID)
+			pullRequestQueue.Remove(prID)
+
+			id := com.StrTo(prID).MustInt64()
+			if _, ok := checkedPRs[id]; ok {
```

Comment:
This is unchanged code, but I think this is incorrect. It basically says we won't recheck any PR that is checked on startup.

Instead of line 166 running the tests this should just add everything to the queue - likely in another goroutine to avoid blocking.